.github/workflows/PR_workflow.yaml

```diff
@@ -0,0 +1,42 @@
+# ------------------------------------------------------------
+# Copyright (c) Microsoft Corporation and Dapr Contributors.
+# Licensed under the MIT License.
+# ------------------------------------------------------------
+
+name: PR_workflow
+
+on:
+  pull_request:
+    branches:
+      - master
+      - release-*
+jobs:
+  build:
+    name: Build ${{ matrix.target_os }}_${{ matrix.target_arch }} binaries
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macOS-latest]
+        target_arch: [amd64]
+        include:
+          - os: macOS-latest
+            target_os: darwin
+    steps:
+      - name: Setup and build
+        if: github.event_name == 'pull_request'
+        run: |
+          brew help
+          DAPR_CLI_VERSION=$( curl -s -f "https://x-access-token:${GITHUB_TOKEN}@api.github.com/repos/dapr/cli/releases" | grep \"tag_name\" | awk 'NR==1{print $2}' |  sed -n 's/\"v\(.*\)\",/\1/p' )
+          echo "Found version ${DAPR_CLI_VERSION}"
+          echo "DAPR_CLI_VERSION=$DAPR_CLI_VERSION" >> $GITHUB_ENV
```

Comment:
No need to set env variables via GITHUB_ENV because they are not used outside this step.

Reply:
they are used inside the generate scripts which are used in the next line